pkg/securitypolicy/securitypolicy.go

```diff
@@ -14,25 +16,64 @@ const (
 	EnvVarRuleRegex  EnvVarRule = "re2"
 )
 
+// PolicyConfig contains toml or JSON config for security policy.
+type PolicyConfig struct {
+	AllowAll   bool              `json:"allow_all" toml:"allow_all"`
+	Containers []ContainerConfig `json:"containers" toml:"container"`
+}
+
+// AuthConfig contains toml or JSON config for registry authentication.
+type AuthConfig struct {
+	Username string `json:"username" toml:"username"`
+	Password string `json:"password" toml:"password"`
+}
+
+// EnvRuleConfig contains toml or JSON config for environment variable
+// security policy enforcement.
+type EnvRuleConfig struct {
+	Strategy EnvVarRule `json:"strategy" toml:"strategy"`
+	Rule     string     `json:"rule" toml:"rule"`
+}
+
+// ContainerConfig contains toml or JSON config for container described
+// in security policy.
+type ContainerConfig struct {
+	ImageName string     `json:"image_name" toml:"image_name"`
+	Command   []string   `json:"command" toml:"command"`
+	Auth      AuthConfig `json:"auth" toml:"auth"`
+	EnvRules  []EnvRule  `json:"env_rules" toml:"env_rule"`
+}
+
+// NewContainerConfig creates a new ContainerConfig from the given values.
+func NewContainerConfig(imageName string, command []string, envRules []EnvRule, auth AuthConfig) ContainerConfig {
```

Comment:
Should this just take in a variadic number of args so we can add new fields if needed without making it a breaking change? The classic `WithCommand("some string")` pattern

Reply:
If there is a change, it is a breaking change. The question would be, is this a compile time error or a runtime error.

I prefer like this where it is a compile time error so if we add additional constraints that are required, then when someone like Sergio who is a consumer of this tries to use without those additional constraints, they get an error at compile time rather than a runtime error when the policy is used.

Reply:
I suppose making things explicit makes more sense for this scenario really